app/models/miq_queue.rb

```diff
@@ -545,6 +545,10 @@ def check_for_timeout(log_prefix = "MIQ(MiqQueue.check_for_timeout)", grace = 10
     end
   end
 
+  def self.check_for_timeout
+    where(:state => STATE_DEQUEUE).find_each(&:check_for_timeout)
```

Comment:
Could find_each be an issue here since this is a high volume table?

Additionally, not sure how possible it is, but can we query by timestamp instead, and only bring back the potential timed out rows?

Reply:
What if I pulled out the logging bit from `#check_for_timeout` into an instance method `#log_timeout` or something then did this:

```ruby
where("updated_on < ?", Time.now.utc - 10.minutes - 10.seconds).destroy_all.each { |msg| msg.log_timeout }
```'

Or I could not do the destroy here and call `#check_for_timeout` as is on those messages.... Thoughts?

Reply:
I guess this doesn't work as well with the message timeout being configurable in the record itself... Will try to find a way to deal with that.

Reply:
Okay I could do this ...
```ruby
where("select date_part('epoch', updated_on) + msg_timeout + 10) < (select date_part('epoch', now())").each(&:check_for_timeout)
```

Reply:
Also, to be fair, we were doing this before as well (https://github.com/ManageIQ/manageiq/pull/19585/files#diff-57a2568ae37208b39c60baa82b4c3a9cL45), but on every server, every 15 seconds.

Reply:
I agree with @carbonin's last point.  We were already doing this on every server previously without scoping it to the server.  Here we're doing it on only one server for the whole database so it's at least (n-1) times less hits on the miq_queue table.

I'm good with doing something like your "ruby" way: 
`where("updated_on < ?", Time.now.utc - 10.minutes - 10.seconds)`

I did enjoy the SQL way though.  :neckbeard:  +10 points

Reply:
@jrafanie We *can't* do it the Ruby way, because the 10 minutes is invalid.  For example, if a job is changed to timeout after an hour, the Ruby query will only look at the ones from the last 10 minutes and never time it out.  We have to either use the "biggest" `msg_timeout` in the table, or just do the SQL way.

>  we were doing this before as well

Yeah, true, and we can leave it for this PR if needed, but I figured if we're in here cleaning it up for performance, why not go a little further.  I'd be ok with a follow up PR though.